### ISSUE TITLE
NET-1738 New Rule S1488: Local variables should not be declared and then immediately returned or thrown

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#7404]
[S1488](https://sonarsource.github.io/rspec/#/rspec/S1488)

Declaring a variable only to immediately return or throw it is a bad practice.

Some developers argue that the practice improves code readability, because it enables them to explicitly name what is being returned. However, this variable is an internal implementation detail that is not exposed to the callers of the method. The method name should be sufficient for callers to know exactly what will be returned.

```csharp
public long ComputeDurationInMilliseconds()
{
  long duration = (((hours * 60) + minutes) * 60 + seconds ) * 1000 ; // Noncompliant
  return duration;
}

public void DoSomething()
{
  ApplicationException myException = new ApplicationException(); // Noncompliant
  throw myException;
}
```

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1738](https://sonarsource.atlassian.net/browse/NET-1738)

[NET-1738]: https://sonarsource.atlassian.net/browse/NET-1738?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
